cfme/utils/appliance/__init__.py

```diff
@@ -2135,45 +2144,71 @@ def check_domain_enabled(self, domain):
         except IndexError:
             raise KeyError("No such Domain: {}".format(domain))
 
-    def configure_openldap(self):
+    @logger_wrap('Configuring openldap external auth provider')
+    def configure_openldap(self, ipaddress, hostname, ldap_conf, sssd_conf,
+                           cert_filepath=None, cert_filename=None, **kwargs):
         """This method changes the /etc/sssd/sssd.conf and /etc/openldap/ldap.conf files to set
             up the appliance for an external authentication with OpenLdap.
             Apache file configurations are updated, for webui to take effect.
-
+            Args:
+                ipaddress: ip of the openldap server
+                hostname: hostname of the openldap server
+                cert_filepath: optional path for placing TLS/SSL cert on appliance
+                cert_filename: optional name of the cert file in conf_path
+                kwargs: additional args for UI auth configuration (3 booleans)
         """
-        ldap_yaml = conf.cfme_data.auth_modes.ext_openldap
+        # TODO refactor using passed auth provider object instead of passed fields
         # write /etc/hosts entry for ldap hostname  TODO DNS
-        self.ssh_client.run_command('echo "{}    {}" >> /etc/hosts'
-                                    .format(ldap_yaml.ipaddress, ldap_yaml.hostname))
+        self.ssh_client.run_command('echo "{}  {}" >> /etc/hosts'
```

Comment:
is Matt working on adding our ldap to dns ?

Reply:
Yes - new auth servers that we deploy will be in DNS. This is an additional configuration to ensure that the given host is resolvable even if not in DNS. We could dynamically decide to add this based on whether or not hostname is resolvable on the appliance though.